urldf54784279_tonyj321_Root_Play_tgz-pJ8-hep_io_root_output_demo_HistogramDemoJ8/src/hep/io/root/output/StreamerFieldInfo.java

```diff
@@ -0,0 +1,171 @@
+package hep.io.root.output;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import hep.io.root.output.annotations.Counter;
+import hep.io.root.output.annotations.FieldType;
+import hep.io.root.output.annotations.Super;
+import hep.io.root.output.annotations.Title;
+
+/**
+ * Summarizes the information known about a field. This information comes from
+ * reflection and annotations.
+ *
+ * @author tonyj
+ */
+class StreamerFieldInfo {
+
+    private final StreamerClassInfo parentClassInfo;
+    private final Field field;
+    private final String title;
+    private final boolean isBase;
+    private final boolean isArray;
+    private final String counter;
+    private final Class fClass;
+    private final StreamerClassInfo fieldClassInfo;
+    private Type type;
+
+    StreamerFieldInfo(StreamerClassInfo c, Field f) throws StreamerInfoException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 38-49, lines 50-59 into separate methods.